dags/web_preaggregated.py

```diff
@@ -0,0 +1,459 @@
+from datetime import datetime, timedelta
+from collections.abc import Callable
+
+import dagster
+from dagster import Field, Array, Definitions
+from clickhouse_driver import Client
+
+from dags.common import ClickhouseClusterResource
+from posthog.models.web_preaggregated.sql import (
+    WEB_OVERVIEW_METRICS_DAILY_SQL,
+    DISTRIBUTED_WEB_OVERVIEW_METRICS_DAILY_SQL,
+    WEB_STATS_DAILY_SQL,
+    DISTRIBUTED_WEB_STATS_DAILY_SQL,
+    WEB_OVERVIEW_INSERT_SQL,
+    WEB_STATS_INSERT_SQL,
+)
+from posthog.clickhouse.cluster import ClickhouseCluster
+
+# We're using 1M pageviews as the desired batch size, this is arbitrary and can be tuned
+# but it is a way to split teams by volume instead of grouping them arbitrarily
+DEFAULT_PAGEVIEW_VOLUME_PER_BATCH = 1_000_000
+
+WEB_ANALYTICS_DATE_PARTITION_DEFINITION = dagster.WeeklyPartitionsDefinition(
+    start_date="2025-01-01",
+    fmt="%Y-%m-%d",
+    timezone="UTC",
+)
+
+WEB_ANALYTICS_CONFIG_SCHEMA = {
+    "team_ids": Field(
+        Array(int),
+        default_value=[],  # Intentionally empty to process all teams but we can use a custom launchpad on dagster to process specific teams while testing
+        description="List of team IDs to process - leave empty to process all teams :fire:",
+    ),
+    "clickhouse_settings": Field(
+        str,
+        default_value="max_execution_time=240, max_bytes_before_external_group_by=21474836480, distributed_aggregation_memory_efficient=1",
```

Comment:
@PostHog/clickhouse This should be half the memory limit currently set when running queries on the production cluster. As mentioned in Slack, I would still face timeouts or memory limits when trying to run those queries for every team (or even a couple of large teams for a month's worth of data), so I am not sure if there is any other setting that would allow us to run it for everyone at once. Still, I am okay with splitting it into batches.